test/wdio_functional/page_objects/common/base_page.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { testSubjectifySelector } from '../../helpers/helpers';
+import Web from './web';
+
+export default class BasePage extends Web {
+
+  constructor(driver) {
+    super(driver);
+    this.driver = driver;
+  }
+
+  /**
+   * @function nav
+   * @returns {Object} Returns an instance of the class that corresponds to the clicked link.
+   */
+  get nav() {
+    const self = this;
```

Comment:
@cjcenizal This didn't work. I think it's because the function creates a new class that I can't use the lexical scope provided by the arrow function. I could be wrong though. Can you get that to work?

Reply:
I actually question if a getter makes sense here, since every time you access the `.nav` property it's going to create a new instance of the Navigation. I would much prefer the Navigation class be constructed in the constructor and the page object/this be passed as an argument:


```js
class BasePage {
  constructor() {
    this.nav = new Navigation(this);
  }
}

class Navigation {
  constructor(pageObject) {
    this.pageObject = pageObject;
  }

  navigateToConsole() {
    this.pageObject.logger.debug
    // ...
  }
}
```